### ISSUE TITLE
[BUG] Saving repeat prefs in FinWin27

### ISSUE THREAD

**Issue by asherber** [finale-lua/lua-scripts#480]
I don't know if this is a PDK Framework bug or a Finale bug. I'm running Finale 27.3.

1. Open a new default document
2. Run the following Lua script
    ```lua
    local prefs = finale.FCRepeatPrefs()
    prefs:Load(1)
    prefs:SetDotAfterNumber(false)
    prefs:Save()
    ```
3. Select the SmartShape tool
4. Go to SmartShape Options
5. In the Symbols dropdown, select Trill
6. Click the "Select..." button
7. CRASH

Using "Save Document Options as JSON" (after step 2), I can see that the values for several of the SmartShape symbols have been changed to impossible values, even though the code that was run didn't touch those preferences (in `FCMusicCharacterPrefs`). 

```json
# These two are okay
"SymbolOctave": 58641,
"SymbolOctaveBass": 58642,
# These four are not
"SymbolTrill": 4294960486,
"SymbolTwoOctaves": 4294960405,
"SymbolTwoOctavesBass": 4294960406,
"SymbolWiggle": 4294961828,
```
I have no idea whether there are other PDK settings that trigger unintended effects; I happened to notice this while importing my house style into a default doc with my "Import Document Options from JSON" script, and then I narrowed the issue down (in my case) to this single setting. Based on the particular values, I feel like this might have something to do with a uint32 overflow, but I have no idea why this repeat setting would affect these four music character settings.

@rpatters1 Are you able to see what's going on here, and to determine whether it's a PDK issue or a Finale issue?

**Comment by asherber** [comment]
Note: This doesn't happen in Fin26, and I haven't tested yet in Fin27 with a non-SMuFL template.

**Comment by rpatters1** [comment]
I have a strong suspicion what's going on. It's difficult to assign blame, but it is certain that if it is going to be fixed it will be fixed in the PDK. I fixed several of these that Jan Angermüller found, but apparently that one slipped by.

If it's what I think it is, you need a minimum of F26.3 and SMuFL (which effectively means F27). If you discover any others, let me know.

**Comment by rpatters1** [comment]
For background:

Finale 26.2 included a complete rewrite of document preferences, including how they are accessed through the PDK. This introduced two major breaking changes in Finale:

- All plugins that referenced document settings would have to be recompiled (and rewritten) to use the new data structures.
- All library files of document settings (think house style libraries) would no longer be usable. They would have to be re-created in F26.2.

This impacted JW plugins in a major way, because by then they had gone out of support by Jari. The other 3rd party plugins made the switch. That's why I create 2 binaries for my plugins: one for F26.2+ and one for F25.4-F26.1. (Let's forget about F25-F25.3: that was another issue.)

Late in the dev cycle they apparently realized these breaking changes were not acceptable. They came up with a compatibility layer for pre-F26.2 libraries that was able to be extended to plugins. The problem is that it is not 100% compatible, especially if your code changes the settings.

Because RGP Lua is a single binary that supports all 64-bit versions, I have only converted it to the new access as needed. (It's a significant job even for one class.) So I will be converting FCRepeatPrefs to the new way as part of 0.67.

FWIW: Jan alerted me to a similar problem with `FCTupletPrefs` that trashed exactly the same character settings as this one does.

Finally, I encourage you to try out the [Enigma Text Dump](https://robertgpatterson.com/-fininfo/-downloads/download-free.html) plugin. It dumps the raw Enigma data into a text file and lets you do A/B comparisons before and after changes. It also includes hints into the PDK Framework where possible. You might have found it useful in debugging this issue. For using the PDK, it is absolutely indispensable. The format is similar the old portable enigma format that existed before F2006.

**Comment by asherber** [comment]
Thanks very much, Robert!

**Comment by rpatters1** [comment]
This will be fixed in 0.67, so I'm closing it now.

**Comment by rpatters1** [comment]
I also added ShowOnTopStaffOnly and ShowOnStaffListID properties to `FCRepeatPrefs`. I'm not sure if these were available elsewhere, but they will be available on `FCRepeatPrefs` starting with 0.67.
